src/main/java/games/strategy/engine/framework/ui/NewGameChooserModel.java

```diff
@@ -38,15 +44,41 @@
     SUCCESS, ERROR
   }
 
-  NewGameChooserModel() {
-    final Set<NewGameChooserEntry> parsedMapSet = parseMapFiles();
+  NewGameChooserModel(final Runnable doneAction) {
+    this(doneAction, NewGameChooserModel::parseMapFiles);
+  }
 
-    final List<NewGameChooserEntry> entries = new ArrayList<>(parsedMapSet);
-    Collections.sort(entries, NewGameChooserEntry.getComparator());
+  /**
+   * Searches for and parses Map Files.
+   */
+  NewGameChooserModel(final Runnable doneAction, final Supplier<Set<NewGameChooserEntry>> mapSupplier) {
+    Preconditions.checkState(SwingUtilities.isEventDispatchThread());
 
-    for (final NewGameChooserEntry entry : entries) {
-      addElement(entry);
-    }
+    final SecondaryLoop loop = Toolkit.getDefaultToolkit().getSystemEventQueue().createSecondaryLoop();
+
+    new SwingWorker<Void, Void>() {
+      @Override
+      protected Void doInBackground() {
+        final List<NewGameChooserEntry> entries = new ArrayList<>(mapSupplier.get());
+        Collections.sort(entries);
+
+        for (final NewGameChooserEntry entry : entries) {
+          SwingUtilities.invokeLater(() -> addElement(entry));
```

Comment:
This is kind of an anti-pattern when using `SwingWorker`.  Idiomatic usage would be to return `entries` as the task return value, and then call `addElement()` on the entries in `done()`.